model/context.go

```diff
@@ -107,10 +107,37 @@ func truncate(s string) string {
 	return s
 }
 
+type NetworkInfo struct {
+	EffectiveType *string
+	RoundTripTime *int64
+	Downlink      *int64
+	DownlinkMax   *int64
+	SaveData      *bool
+	Type          *string
+}
+
+func (ni *NetworkInfo) fields() common.MapStr {
+	fields := common.MapStr{}
+	if ni == nil {
+		return fields
+	}
+	utility.Set(fields, "type", ni.Type)
+	utility.Set(fields, "downlink", ni.Downlink)
+	utility.Set(fields, "downlinkMax", ni.DownlinkMax)
+	utility.Set(fields, "effectiveType", ni.EffectiveType)
+	utility.Set(fields, "roundTripTime", ni.RoundTripTime)
+	utility.Set(fields, "saveData", ni.SaveData)
```

Comment:
I don't see "savedata" in the [Network Information API](https://wicg.github.io/netinfo/#networkinformation-interface) spec. Seems to me it's more a property of the request (page) than the network information.

@vigneshshanmugam where does this property come from?

Reply:
You are right, the spec network Information API got updated and it no longer has `saveData` but its supported by chrome and might be added in future so we can spec it and remove usage if not required. 
<img width="275" alt="Screenshot 2020-06-26 at 10 47 47" src="https://user-images.githubusercontent.com/3902525/85838879-7e4c5600-b79a-11ea-92da-a25d67cc7590.png">

Reply:
Preview is here - https://s3.amazonaws.com/pr-preview/WICG/netinfo/saveData.html